client/src/main.rs

```diff
@@ -466,6 +466,7 @@ fn main() -> Result<()> {
         key_pem: opt.key_pem,
         root_cert_pem: opt.root_cert_pem,
         control: opt.control,
+        read_only: false,
```

Comment:
What are your thoughts on making "read_only" a field of the `Upstairs` struct?
I wonder if making that (and perhaps even lossy) a struct member so anyone can look them up.  Then we don't have to pass that state around everywhere?